src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -75,18 +77,18 @@ public CommandResult execute(Model model) throws CommandException {
         Optional<Person> personOptional = findPersonToEdit(lastShownList);
 
         if (personOptional.isEmpty()) {
+            logger.log(Level.WARNING, "Person not found for editing");
             throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
         }
 
         Person personToEdit = personOptional.get();
+        logger.log(Level.INFO, "Editing person: {0}", personToEdit);
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
-        }
-
```

Comment:
Was this removed because we cannot edit the identifier fields of a `Person`?

Reply:
Yes. Since we are not able to edit a person's Name or Nric, there will never be a case of duplicate persons in the address book as a result of using the editCommand.

Reply:
OK. There are minor redundancies for `EditPersonDescriptor` in this case, but this solution works regardless of that.